src/app/shared/navigation-focus/navigation-focus.service.ts

```diff
@@ -0,0 +1,43 @@
+import {ElementRef, Injectable} from '@angular/core';
+import {NavigationEnd, Router} from '@angular/router';
+import {filter, skip, take} from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class NavigationFocusService {
+  private navigationFocusRequests: ElementRef[] = [];
+  private skipLinkFocusRequests: ElementRef[] = [];
+
+  constructor(private router: Router) {
+    this.router.events
+      .pipe(filter(event => event instanceof NavigationEnd),
+        take(1))
+      .subscribe(() => {
+        setTimeout(() =>
+          this.skipLinkFocusRequests[this.skipLinkFocusRequests.length - 1].nativeElement.id =
```

Comment:
I don't think we need to wait for a navigation request to do this, we can just do it right when they request.

Also, I would prefer if we have an id on the element and then update the skiplink to refer to that ID, rather than changing the ID on the element